workers/unity/Packages/com.improbable.gdk.transformsynchronization/Systems/ResetForAuthorityGainedSystem.cs

```diff
@@ -52,59 +52,56 @@ protected override void OnUpdate()
             UpdateTransformData();
         }
 
-        private void UpdateTransformData()
+        private void UpdateRigidbodyData()
         {
-            var ticksSinceLastUpdateArray = rigidbodyGroup.GetComponentDataArray<TicksSinceLastTransformUpdate>();
-            var transformComponentArray = rigidbodyGroup.GetComponentDataArray<TransformInternal.Component>();
-            var bufferedTransformArray = rigidbodyGroup.GetBufferArray<BufferedTransform>();
-            var rigidbodyArray = rigidbodyGroup.GetComponentArray<Rigidbody>();
-            var spatialEntityIdArray = rigidbodyGroup.GetComponentDataArray<SpatialEntityId>();
-
-            for (int i = 0; i < transformComponentArray.Length; ++i)
-            {
-                // todo this is not a correct constraint. Needs a the auth loss temporary exposed to correctly do this
-                // alternatively this needs an authority changed component that is filled at the beginning of the tick
-                if (updateSystem
-                    .GetAuthorityChangesReceived(spatialEntityIdArray[i].EntityId, TransformInternal.ComponentId)
-                    .Count == 0)
+            Entities.With(rigidbodyGroup).ForEach(
+                (Entity entity, DynamicBuffer<BufferedTransform> buffer,
+                    ref TicksSinceLastTransformUpdate ticksSinceLastTransformUpdate,
+                    ref TransformInternal.Component transformInternal,
+                    ref SpatialEntityId spatialEntityId) =>
                 {
-                    continue;
-                }
+                    // todo this is not a correct constraint. Needs a the auth loss temporary exposed to correctly do this
```

Comment:
ping on this as well. if ticket was created, please link here

Reply:
@samiwh I have no idea what this is

Reply:
If authority was temporarily lost (so no auth change per tick, but another worker could have changed the position) then the ridigbody needs to be reset. As this only checks for the auth change it will miss that case.

Reply:
could you please make a ticket :D